apps/web/src/pages/integrations/components/ConnectIntegrationForm.tsx

```diff
@@ -110,6 +122,20 @@ export function ConnectIntegrationForm({
     }
   >(({ integrationId, data }) => updateIntegration(integrationId, data));
 
+  const { data: webhookSupportStatus } = useQuery(
+    ['webhookSupportStatus', provider?.channel, provider?.providerId],
+    () =>
+      getWebhookSupportStatus({
+        channel: provider?.channel as ChannelTypeEnum.EMAIL | ChannelTypeEnum.SMS,
+        providerId: provider?.providerId as string,
+      }),
+    {
+      enabled: Boolean(
+        provider?.providerId && [ChannelTypeEnum.EMAIL, ChannelTypeEnum.SMS].includes(provider.channel) && !createModel
+      ),
+    }
```

Comment:
Mmm... now I notice, maybe you wouldn't want to run the endpoint for some channels. This would conflict with my suggestions though we could do a list of allowed providers to run this endpoint.

Reply:
Yes, the endpoint checks if the given provider supports webhook or not
It checks the availability of the `parseEventsBody` method on the given provider to confirm webhook support